demo/exercises/src/main/java/com/khoaquang/garbagecollector_memoryleaks/exercise2/MemoryLeaks.java

```diff
@@ -0,0 +1,58 @@
+package com.khoaquang.garbagecollector_memoryleaks.exercise2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Behavior:
+ *   This program creates a lot of Task objects, and only a small number are removed, while the rest stay in memory.
+ *   Over time, the heap memory will fill up, leading to OutOfMemoryError in large-scale applications.
+ * How to Fix the Memory Leak:
+ *   To prevent the memory leak, ensure that objects are properly dereferenced when no longer needed. For example:
+ *   taskList.clear(); // Clears all references in the list
+ */
+public class MemoryLeaks {
+    static class Task {
+        private final byte[] data = new byte[1024 * 1024]; // 1 MB data
+        private final String name;
+
+        Task(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public String toString() {
+            return name;
+        }
+    }
+
+    public static void main(String[] args) {
+        List<Task> taskList = new ArrayList<>();
+        System.out.println("Adding tasks to the list...");
```

Comment:
Exercise 2: Create a memory leak example (e.g., using unclosed resources or static references). Document the leak’s impact and steps to fix it.

I aslo think this code don't match with the description above

Reply:
Why do you think it doesn't match with the exercise?

Reply:
It's ok